sleap_nn/training/model_trainer.py

```
@@ -68,38 +86,112 @@
+        if not self.config.trainer_config.save_ckpt_path:
+            self.dir_path = "."
+        else:
+            self.dir_path = self.config.trainer_config.save_ckpt_path
+
+        if not Path(self.dir_path).exists():
+            try:
+                Path(self.dir_path).mkdir(parents=True, exist_ok=True)
+            except OSError as e:
+                raise OSError(
+                    f"Cannot create a new folder. Check the permissions to the given Checkpoint directory. \n {e}"
+                )
```

Ruff

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)